pkg/util/net/ip/ip.go

```diff
@@ -0,0 +1,81 @@
+/*
+ * This file is part of the KubeVirt project
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * Copyright 2019 Red Hat, Inc.
+ *
+ */
+package ip
+
+import (
+	"bufio"
+	"net"
+	"os"
+)
+
+const (
+	// sysfs
+	DISABLE_IPV6_PATH = "/proc/sys/net/ipv6/conf/default/disable_ipv6"
+
+	// Special IPs
+	IPV4_BIND_ADDRESS     = "0.0.0.0" //INADDR_ANY
+	IPV4_LOOPBACK_ADDRESS = "127.0.0.1"
+	IPV6_BIND_ADDRESS     = "[::]" //INADDR6_ANY
+	IPV6_LOOPBACK_ADDRESS = "[::1]"
+)
+
+func readSysfsFile(path string) (string, error) {
+	var output string = ""
+	file, err := os.Open(path)
+	if err != nil {
+		return output, err
+	}
+
+	defer file.Close()
+	scanner := bufio.NewScanner(file)
+	if scanner.Scan() {
+		output = scanner.Text()
+	}
+	if err := scanner.Err(); err != nil {
+		return output, err
+	}
+
+	return output, nil
+}
+
+// GetIPBindAddress returns IP bind address (either 0.0.0.0 or [::] according sysctl disable_ipv6)
+func GetIPBindAddress() string {
+	disableIPv6, err := readSysfsFile(DISABLE_IPV6_PATH)
+	if err != nil || disableIPv6 == "1" {
+		return IPV4_BIND_ADDRESS
+	}
+
+	return IPV6_BIND_ADDRESS
+}
+
+// GetLoopbackAddress returns loopback IP (either 127.0.0.1 or [::1] according sysctl disable_ipv6)
+func GetLoopbackAddress() string {
+	disableIPv6, err := readSysfsFile(DISABLE_IPV6_PATH)
+	if err != nil || disableIPv6 == "1" {
+		return IPV4_LOOPBACK_ADDRESS
+	}
+
+	return IPV6_LOOPBACK_ADDRESS
+}
+
+// IsLoopbackAddress checks if the address is IPv4 / IPv6 loopback address
+func IsLoopbackAddress(ipAddress string) bool {
+	loopback := net.ParseIP(ipAddress)
+	return loopback.IsLoopback() || ipAddress == IPV6_LOOPBACK_ADDRESS
```

Comment:
I cannot find my previous comment and your response. You said you cannot use just loopback.IsLoopback since the ipv6 address contains the brackets. As Eddy mentioned in the comment (https://github.com/kubevirt/kubevirt/pull/3392/commits/fde14127e7e5b3035167a1a7a726c9a387fa7fa8#r422618271). Adding the brackets should be done most close as possible to creating the socket.
IMO,
* `GetIPBindAddress` and `GetLoopbackAddress` should return a raw address. Without the brackets.
* The `tcpBindAddress` passed to `NewTargetProxy` shouldn't have brackets. Add the brackets when you need them, and that's when creating a socket.
* BTW, with your current code, having the brackets in the `tcpBindAddress` the code of [PR 3396](https://github.com/kubevirt/kubevirt/pull/3396/commits/d5fd692c7afde5e2935074a269a4e31d66d063f0#diff-e88e0fbecb2c3ab1c8c08d12a1c8b4f6R220) is redundant.

Reply:
This is what i did in the beginning and then was asked to move from there,
anyway i will try and see what it involves, i assume you want it as part of this PR,
just saying we already said lgtm in the other one, and decided it will be in other PR...

Reply:
About "GetIPBindAddress and GetLoopbackAddress should return a raw address. Without the brackets." it might add more places that need normalization, i prefer if we can stick to the original plan and please do it in a separate PR ?
I already started couple other things, and also this investigation, but planning and order is better when possible please.

Reply:
keep in mind please that the condition i added can be easily removed if i don't use the golib, but have the two comparison, and then the code would be great, and the 2nd issue Edy and you raised can be fixed in a following PR imo.

https://github.com/kubevirt/kubevirt/pull/3396/files#diff-4713d8334b0e7a78d689e3a725ab753fR80